### ISSUE TITLE
AIP-127: Update HTTP Template Syntax rule disallows hyphens in paths

### ISSUE THREAD

**Issue by apodznoev** [googleapis/api-linter#1427]
The current AIP Linter implementation enforces a rule that disallows hyphens (-) as literals within HTTP Pattern Variables. 

This can lead to issues when designing API paths that use hyphens for readability, such as:
```
service Library {
  rpc GetMyBooks(GetMyBooksRequest) returns (Book) {
    option (google.api.http) = {
      get: "/v1/my-books"
    };
  }
}
``` 

This example is flagged as invalid by the linter, even though hyphens are considered safe characters according to the LITERAL syntax definition in:
https://github.com/googleapis/googleapis/blob/83c3605afb5a39952bf0a0809875d41cf2a558ca/google/api/http.proto#L224

> The syntax `LITERAL` matches the literal text in the URL path. If the `LITERAL`
> contains any reserved character, such characters should be percent-encoded
> before the matching.
> 

Since hyphens do not require encoding and are commonly used in URL construction, there seems to be no technical justification for the linter to restrict their use.

**Comment by noahdietz** [comment]
Hi, thanks for the issue. The linter is enforcing Google style here not necessarily the exact allowable syntax of an HTTP path. We use lowerCamelCase for multi-word segments. Feel free to disable this rule in your own configuration or even fork it and register your own rule to make `-` allowable for your use cases.

> rule that disallows hyphens (-) as literals within HTTP Pattern Variables

A bit of a pedantic nit: The example isn't a pattern variable, but a literal segment - `{book_id}` would be a variable segment - and when a variable segment is expanded with a value at runtime that may contain a `-` as described in the
